src/components/FeatureViews/Courses/CourseManagement.js

```diff
@@ -0,0 +1,490 @@
+  const startingDate = moment(startDate).calendar();
+  const endingDate = moment(endDate).calendar();
+
+  const isActive = (startTime, endTime, startDate, endDate) => {
+    const currentTime = moment().format("HH:mm:ss");
+    const currentDate = moment().format("L");
+    return (
+      currentDate >= startDate &&
+      currentTime >= startTime &&
+      currentTime <= endTime &&
+      currentDate <= endDate
+    );
+  };
+
+  const handleClick = (e) => {
+    history.push(`/coursemanagement/class/${id}`);
+  };
+
+  return (
+    <>
+      <Grid
+        container
+        justify="flex-start"
+        className={classes.mainCardContainer}
+        data-active="inactive"
+        onClick={handleClick}
+      >
+        <Grid item xs={6} sm={9} md={6}>
+          <Typography variant="h4" align="left" style={{ marginLeft: ".85em" }}>
+            {title}
+          </Typography>
+        </Grid>
+        <Grid item xs={6} sm={3} md={6} style={{ textAlign: "left" }}>
+          <Chip
+            label={
+              isActive(startTime, endTime, startingDate, endingDate)
+                ? "ACTIVE"
+                : "PAST"
+            }
+            className={classes.chipSize}
+            style={{
+              backgroundColor: isActive(
+                startTime,
+                endTime,
+                startingDate,
+                endingDate
+              )
+                ? activeColor
+                : pastColor,
+            }}
+          />
+        </Grid>
+        <Grid item xs={3} sm={4} md={3} className={classes.displayCardMargins}>
+          <Typography
+            variant="body1"
+            align="left"
+            style={{ marginLeft: "1.85em" }}
+          >
+            <span style={{ marginRight: theme.spacing(1) }}>By:</span>
+            <span className={classes.highlightName}>{`${concatFullName}`}</span>
+          </Typography>
+        </Grid>
+        <Divder
+          orientation="vertical"
+          flexItem
+          style={{ height: "2em", marginTop: "1em" }}
+        />
+        <Grid item xs={6} sm={7} md={6}>
+          <Typography
+            variant="body1"
+            align="left"
+            style={{ marginLeft: "1.2em", paddingTop: "3px" }}
+            className={classes.displayCardMargins}
+          >{`Time: ${startingDate} - ${endingDate} ${abbreviatedDay} ${startingTime} - ${endingTime} `}</Typography>
+        </Grid>
+      </Grid>
+      <Divder />
+    </>
+  );
+};
+
+const CourseFilter = ({ initialValue, filterList, setState, filter }) => {
+  const classes = useStyles();
+  const handleChange = (event) => setState(event.target.value);
+
+  const chosenFilter = filterList.map((filterItem) => {
+    switch (initialValue) {
+      case "All Instructors":
+        return (
+          <MenuItem
+            key={filterItem.instructor.user.lastName}
+            className={classes.menuSelect}
+            value={filterItem.instructor.user.lastName}
```

Comment:
never make the value of a `<MenuItem>` the name/label of something. this should always be an id so you don't run into a race condition (a case where 2 things conflict). for example, if there are 2 instructors named "Wilson Lam", then you'll have 2 values as  "Lam" which isn't good because each `<MenuItem/>` is supposed to have a unique value.